node/src/tests/inherent_data_tests.rs

```diff
@@ -90,27 +94,29 @@ async fn block_proposal_cidp_should_be_created_correctly() {
 
 #[tokio::test]
 async fn block_verification_cidp_should_be_created_correctly() {
-	let mut block_data_source = MockBlockDataSource::default();
-	let parent_stable_block = block_data_source.get_all_stable_blocks().first().unwrap().clone();
+	let parent_stable_block = MainchainBlock {
+		number: McBlockNumber(1),
+		hash: McBlockHash([1; 32]),
+		epoch: McEpochNumber(2),
+		slot: McSlotNumber(3),
+		timestamp: 4,
+	};
 	let mc_block_hash = McBlockHash([2; 32]);
-	block_data_source.push_stable_block(MainchainBlock {
+	let native_token_data_source = MockNativeTokenDataSource::new(
+		[((None, mc_block_hash.clone()), NativeTokenAmount(1000))].into(),
+	);
+	let mc_hash_data_source = MockMcHashDataSource::from(vec![MainchainBlock {
 		number: McBlockNumber(parent_stable_block.number.0 + 5),
 		hash: mc_block_hash.clone(),
 		slot: McSlotNumber(parent_stable_block.slot.0 + 100),
 		timestamp: parent_stable_block.timestamp + 101,
 		epoch: McEpochNumber(parent_stable_block.epoch.0),
-	});
-	let native_token_data_source = MockNativeTokenDataSource::new(
-		[(
-			(None, block_data_source.stable_blocks.last().unwrap().hash.clone()),
-			NativeTokenAmount(1000),
-		)]
-		.into(),
-	);
+	}]);
 	let data_sources = DataSources {
-		block: Arc::new(block_data_source),
 		candidate: Arc::new(MockCandidateDataSource::default()),
 		native_token: Arc::new(native_token_data_source),
+		mc_hash: Arc::new(mc_hash_data_source),
+		sidechain_rpc: Arc::new(SidechainRpcDataSourceMock::new(parent_stable_block)),
```

Comment:
Does inherent data provider need `sidechain_rpc` data source? I think it should not.

Reply:
You're right. I changed the CIDPs to only take the data sources they need.